contracts/RadiCards.sol

```diff
@@ -3,234 +3,504 @@ pragma solidity ^0.4.24;
+            _logo
+        );
+        benefactorsIndex.push(_benefactorIndex);
+
+        emit BenefactorAdded(_benefactorIndex);
+        return true;
+    }
+
+    function addCard(uint256 _cardIndex, string _tokenURI, bool _active, uint256 _maxQnty, uint256 _minPrice)
+    public onlyIfWhitelisted(msg.sender)
+    returns (bool) {
+        require(bytes(_tokenURI).length != 0, "Invalid token URI");
+
+        cards[_cardIndex] = CardDesign(
+            _tokenURI,
+            _active,
+            0,
+            _maxQnty,
+            _minPrice
+        );
+        cardsIndex.push(_cardIndex);
+
+        emit CardAdded(_cardIndex);
+        return true;
+    }
+
+    function setTokenBaseURI(string _newBaseURI) external onlyIfWhitelisted(msg.sender) {
+        require(bytes(_newBaseURI).length != 0, "Base URI invalid");
+
+        tokenBaseURI = _newBaseURI;
+    }
+
+    function setActive(uint256 _cardIndex, bool _active) external onlyIfWhitelisted(msg.sender) {
+        require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+        cards[_cardIndex].active = _active;
+    }
+
+    function setMaxQuantity(uint256 _cardIndex, uint256 _maxQnty) external onlyIfWhitelisted(msg.sender) {
+        require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+        require(cards[_cardIndex].minted<=_maxQnty, "Cant set the max quantity less than the current total minted");
+        cards[_cardIndex].maxQnty = _maxQnty;
+    }
+
+    function setMinPrice(uint256 _cardIndex, uint256 _minPrice) external onlyIfWhitelisted(msg.sender) {
+        require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+        cards[_cardIndex].minPrice = _minPrice;
+    }
+
+    function setDaiContractAddress(address _daiERC20ContractAddress) external onlyIfWhitelisted(msg.sender){
+        require(_daiERC20ContractAddress != address(0), "Can't be address zero");
+        daiContract = StandardToken(_daiERC20ContractAddress);
+    }
+
+    function setMedianizerContractAddress(address _MedianizerContractAddress) external onlyIfWhitelisted(msg.sender){
+        require(_MedianizerContractAddress != address(0), "Can't be address zero");
+        medianizerContract = Medianizer(_MedianizerContractAddress);
+    }
+
+  // returns the current ether price in usd. 18 decimal point precision used
+    function getEtherPrice() public view returns(uint256){
+        return uint256(medianizerContract.read());
+    }
+
+  //returns the value of ether in atto  (1 usd of ether = 10^18 atto)
+    function getEthUsdValue(uint256 _ether) public view returns(uint256){
+        return ((_ether*getEtherPrice())/(1 ether));
+    }
+
+  //returns the minimum required in wei for a particular card given the card min price in dai
+    function getMinCardPriceInWei(uint256 _cardIndex) public view returns(uint256){
+        return ((cards[_cardIndex].minPrice*1 ether)/getEtherPrice());
```

Comment:
Looks good, alot of functionality has been added and from first look it seem great. I would make you are happy enough with the test coverage and some of the edge cases around claiming etc so we can be confident there isn't a sneaky little bug. Top banana 🔝 🍌